### PR TITLE
Move client-side assets to the `/assets` route

### PR DESCRIPTION
**Pull request by pitaj**
- Route `/assets` points to `build/public` directory
- If file not found in `build/public`, it falls back to `public/`
- Build destinations for templates, minfiles, and sounds moved to `build/public`
- r.js modules previously linked through express, now symlinked to `public/build/src/modules`
- Minfiles no longer served from memory cache
- Use `views_dir` nconf setting everywhere template directory is used
- Has fallbacks for existing themes `/*` redirects to `/assets/*`
- Doesn't move plugin assets into the assets route, those are still at `/plugins`
- Refactor meta/css
- `fs.link` / `fs.symlink` usage abstracted to `file.link`
- Remove `local-assets` flag, it was broken anyways
- Route `/plugins` points to `build/public/plugins` directory
- Plugin `staticDirs` now symlinked into `build/public/plugins`

### CODE REVIEW

**Comment by pitaj** [comment]
Here's a sample nginx config for use with the new routes introduced in this PR:

```
upstream nodes {
    ip_hash;
    server 127.0.0.1:4567;
}

server {
    listen 80;
    server_name $SITE_NAME;

    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Host $http_host;
    proxy_set_header Referer $http_referer;
    proxy_set_header X-NginX-Proxy true;
    proxy_redirect off;


    location @nodebb {
        proxy_pass http://nodes;
    }

    location @public {
        root /home/saas/nodebb/public/;
        try_files $uri $uri/ @nodebb;
    }

    location /assets/ {
        root /home/saas/nodebb/build/public/;
        try_files $uri $uri/ @public;
    }

    location /plugins/ {
        root /home/saas/nodebb/build/public/plugins/;
        try_files $uri $uri/ @nodebb;
    }

    location / {
        proxy_pass http://nodes;
    }
}
```

**Comment by julianlam** [comment]
The existing nginx config block is:

```
    location ~ ^/(images|language|sounds|templates|uploads|vendor|src\/modules|nodebb\.min\.js|stylesheet\.css|admin\.css) {
        root /path/to/nodebb/public/;
        try_files $uri $uri/ @nodebb;
    }
```

Just want to confirm that this won't **break** any existing installs with that in... from what I can tell it should just continue to work (though nginx won't actually be serving any of those assets itself)

**Comment by pitaj** [comment]
Yes, since it passes through, it should continue to work.

File: src/meta/css.js [hunk 1]
```diff
@@ -18,8 +18,32 @@ var utils = require('../../public/src/utils');
...
+			return source + '\n' + [
+				'@import "font-awesome";',
+				'@import "../public/less/admin/admin";',
+				'@import "../public/less/generics.less";',
+				'@import (inline) "../public/vendor/colorpicker/colorpicker.css";',
+				'@import (inline) "../public/vendor/jquery/css/smoothness/jquery-ui.css";',
+				'@import (inline) "../public/vendor/jquery/bootstrap-tagsinput/bootstrap-tagsinput.css";',
+			].map(function (str) { return str.replace(/\//g, path.sep); }).join('\n');
+		},
+	};
```
Comment (julianlam) [inline src/meta/css.js]: Hey, nice touch.

**Comment by julianlam** [comment]
Looks like you've got a test to fix 😬

**Comment by pitaj** [comment]
That's actually a bad test. dbsearch isn't activated by default. A better test would use mentions or markdown or something that's activated by default.

**Comment by julianlam** [comment]
I like how fixing tabs causes your PR to fail coveralls test :trollface:

**Pull request merged**